src/common/utils/deltaUtils.ts

```diff
@@ -0,0 +1,177 @@
+    return { members: languages.map(languageSuffix => sobject + "-" + languageSuffix), name: ["CustomObjectTranslation"] };
+  };
+
+  const objectTranslations = async (member: string): Promise<PackageType> => {
+    const languages = await allLanguages(fullPackageFile);
+    return { members: languages.map(suffix => member + "-" + suffix), name: ["CustomObjectTranslation"] };
+  };
+
+  const globalTranslations = async () => {
+    const languages = await allLanguages(fullPackageFile);
+    return { members: languages, name: ["Translations"] };
+  }
+
+  const allCustomMetadataRecords = async (member: string): Promise<PackageType | null> => {
+    if (!member.includes('__mdt')) {
+      return null;
+    }
+    const baseName = member.split('__mdt')[0];
+    const types = await allTypes(fullPackageFile);
+    const metadataRecords = 
+      (cache.customMetadata ?? (cache.customMetadata = types.find(type => type.name[0] === "CustomMetadata")))
+      ?.members
+      .filter(member => member.startsWith(baseName));
+
+    return metadataRecords?.length ? { members: metadataRecords, name: ["CustomMetadata"] } : null;
+  }
+
+  const allCustomFields = async (member: string): Promise<PackageType | null> => {
+    const baseName = member.split('.')[0];
+    const types = await allTypes(fullPackageFile);
+    const metadataFields = 
+      (cache.customFields ?? (cache.customFields = types.find(type => type.name[0] === "CustomField")))
+      ?.members
+      .filter(field => field.startsWith(baseName));
+
+    return metadataFields?.length ? { members: metadataFields, name: ["CustomField"] } : null;
+  }
+
+  const allObjectRecordTypes = async (member: string): Promise<PackageType | null> => {
+    
+    if (member.includes("__mdt")) {
+      return null;
+    }
+    const parts = member.split('.');
+    if (parts.length !== 2) {
+      return null;
+    }
+    const types = await allTypes(fullPackageFile);
+    const sobject = parts[0];
+
+    const recordTypesMembers =
+      (cache.recordTypes ?? (cache.recordTypes = types.find(type => type.name[0] === "RecordType")))
+      ?.members
+      .filter(member => member.startsWith(sobject + '.'));
+
+    return recordTypesMembers?.length ? { members: recordTypesMembers, name: ["RecordType"] } : null;
+  };
+
+  const metadataProcessors = {
+    "Layout" : dashSeparatedObjectToObjectTranslation,
+    "CustomObject" : objectTranslations,
+    "ValidationRule" : dotSeparatedObjectToObjectTranslation,
+    "QuickAction" : dotSeparatedObjectToObjectTranslation,
+    "RecordType" : dotSeparatedObjectToObjectTranslation,
+    "CustomMetadata" : [allCustomFields],
+    "CustomLabel" : globalTranslations,
+    "CustomPageWebLink" : globalTranslations,
+    "CustomTab" : globalTranslations,
+    "ReportType" : globalTranslations,
+    "CustomField" : [allObjectRecordTypes, allCustomMetadataRecords, dotSeparatedObjectToObjectTranslation],
+  };
+  
+  const xml = await parseXmlFile(deltaXmlFile);

```

Comment:
you also have methods parsePackageXmlFile and writePackageXmlFile , that are more handy :)

Reply:
done